src/backend/gl/src/command.rs

```diff
@@ -1274,10 +1301,42 @@ impl command::RawCommandBuffer<Backend> for RawCommandBuffer {
         &mut self,
         _layout: &n::PipelineLayout,
         _stages: pso::ShaderStageFlags,
-        _offset: u32,
-        _constants: &[u32],
+        offset: u32,
+        constants: &[u32],
     ) {
-        unimplemented!()
+        let buffer = self.add(constants);
+
+        let uniforms = &self.cache.uniforms;
+        if uniforms.is_empty() {
+            unimplemented!()
+        }
+
+        let uniform = if offset == 0 {
+            // If offset is zero, we can just return the first item
+            // in our uniform list
+            uniforms.get(0).unwrap()
+        } else {
+            // If offset is non-zero, we need to iterate through our uniform list
+            // and find what uniform is expected
+            let mut needle = None;
+            let mut acc: u32 = 0;
+            for uniform in uniforms {
+                acc += uniform.size as u32;
+                if acc == offset {
```

Comment:
If we are only going to check for the uniforms' offsets, not their size, then we should only store the uniform offsets, not sizes. Then we can just do a `binary_search`.